_posts/2024-03-14-the-big-rename.adoc

```diff
@@ -0,0 +1,48 @@
+
+Historically Quarkus has used the term "reactive" in the names of several extensions to indicate their *additional* support for non-blocking, event-driven programming. 
+
+However, we've found that this terminology has led to confusion and misunderstanding among our users. Many have mistakenly believed that using these "reactive" extensions forces them into reactive programming, even though Quarkus always by default recommend traditional blocking programming model for your application and only optionally use reactive. 
+
+With the recent production release of Virtual Threads in Java 21, the distinction between traditional, reactive, and now, virtual thread-based programming has become even more critical. 
+
+== Our Solution: A New Naming Strategy
+
+To address this and guide our users more effectively, we've decided to revise the naming convention for certain extensions. This change aims to clarify the capabilities and flexibility of Quarkus extensions, making it easier for you to choose the right extension for your project.
+
+Here's a summary of the changes:
+
+- *Quarkus RESTEasy Reactive* becomes *Quarkus REST*, emphasizing its general-purpose applicability beyond just reactive programming.
+- *Quarkus SmallRye Reactive Messaging* is now *Quarkus Messaging*, indicating its broader use cases.
+- Extensions that inherently push for a reactive programming model, like *Quarkus Hibernate Reactive*, will retain the "reactive" terminology.
+- The *Quarkus REST Client Reactive* is simplified to *Quarkus REST Client*, unifying the naming under a more general "REST" umbrella.
+
+== Impact on users
+
+These changes are designed to make it clearer that you're not constrained to reactive programming when using these extensions. Whether you prefer a traditional blocking approach, reactive programming and/or virtual threads, Quarkus supports you.
+
+For most projects, this renaming should not require changes, except for those having used `quarkus-rest-client` it may require some adjustments in your dependencies. We're committed to making this transition as smooth as possible. Here are a few key points to help you adapt:
+
+- **Maven Relocations:** We'll introduce Maven relocations for the renamed extensions, guiding you towards the new names when you update your dependencies.
+- **Tooling Updates:** The Quarkus CLI will default to use the new names for new projects but continue to work with old names. `quarkus update` will get rules to help perform the update/renames if/when relevant.
+- **Documentation:** We'll update our documentation to reflect the new extension names, making it easier for you to find the right extension for your project.
+
+For current projects we recommend you to keep using the old names if you want to somehow still stay buildable against older version of Quarkus - otherwise upgrade in your own pace for new names.
```

Comment:
```suggestion
For current projects, we recommend continuing to use the old names if you want to remain buildable against an older version of Quarkus — otherwise, upgrade at your own pace for new names.
```